lib/pipefy_message/broker/aws/sns/publisher.rb

```diff
@@ -2,38 +2,45 @@
 
 require "aws-sdk-sns"
 require "json"
+require_relative "../configuration"
 
 module PipefyMessage
-  # Aws SNS Publisher class to publish json messages into a specific topic
-  class SnsPublisher
-    def initialize
-      PipefyMessage::AwsProviderConfig.instance.setup_connection
-      @sns = Aws::SNS::Resource.new
-    end
+  module Publisher
+    module AwsProvider
+      # Aws SNS Publisher class to publish json messages into a specific topic
+      class SnsPublisher
+        def initialize
+          aws_config = PipefyMessage::BrokerConfiguration::AwsProvider::ProviderConfig.instance
+          aws_config.setup_connection
+          @sns = Aws::SNS::Resource.new
+        end
 
-    def publish(payload, topic_arn)
-      message = prepare_payload(payload)
-      do_publish(message, topic_arn)
-    end
+        def publish(payload, topic_arn)
+          message = prepare_payload(payload)
+          do_publish(message, topic_arn)
+        end
 
-    private
+        private
 
-    def prepare_payload(payload)
-      # The 'Default' json key/entry it's mandatory to ruby sdk
-      {
-        "default" => payload
-      }
-    end
+        def prepare_payload(payload)
+          # The 'Default' json key/entry it's mandatory to ruby sdk
+          {
+            "default" => payload
+          }
+        end
 
-    def do_publish(message, topic_arn)
-      topic = @sns.topic(topic_arn)
+        def do_publish(message, topic_arn)
+          topic = @sns.topic(topic_arn)
 
-      puts "Publishing a json message to topic #{topic_arn}"
-      result = topic.publish({
-                               message: message.to_json,
-                               message_structure: "json"
-                             })
-      puts "Message Published with ID #{result.message_id}"
+          puts "Publishing a json message to topic #{topic_arn}"
+          result = topic.publish({
+                                   message: message.to_json,
+                                   message_structure: "json"
+                                 })
+          puts result
+          puts "Message Published with ID #{result.message_id}"
```

Comment:
today we have this return `{:message_id=>"0fa99c3b-d876-4110-aca3-625de62abc4b", :sequence_number=>nil}` from the provider, I think only the message_id will be useful, WDYT?

ps: we can return the whole hash too.

Reply:
@jleber yes I think just the message ID is the important information here, but just returning a hash in the first place is a good start for us rather than returning `nil` (by having the `puts` call as the last line)